### DISCUSSION TITLE
Useful Ubuntu 22.04/24.04 VM Commands

### DISCUSSION THREAD

**Discussion by havardthom** [community-scripts/ProxmoxVE#272] [Guides]
Copied from original post: https://github.com/tteck/Proxmox/discussions/2072

## Setting up Cloud-Init
![image](https://github.com/user-attachments/assets/eac14e5b-6013-4bcc-8b78-602dffa09af1)

1. Set user to root
2. Give root user a password
3. Set to no (not enough storage, yet)
4. Set network settings
5. Regenerate Image

**Now you can start the VM**

### Select xterm.js under the Console pull down for copy/paste functions
![image](https://github.com/user-attachments/assets/32ce91c3-ea63-47de-aaa4-86821c190d17)

### Hardware > Hard Disk (scsi0) > Disk Action > Resize
![image](https://github.com/user-attachments/assets/e55da74d-5684-457d-98a0-9b99eff85dd3)

## Expand VM Disk using parted (/dev/sda1)
`parted /dev/sda`
`resizepart 1`
Fix/Ignore? `Fix`
Partition number? `1`
Yes/No? `Yes`
End? [2146MB]? `-0`
(parted) `quit`
**reboot**

## Get SSH Going
```
sed -i -e 's/#PermitRootLogin prohibit-password/PermitRootLogin yes/g' -e 's/^PasswordAuthentication.*/PasswordAuthentication yes/' /etc/ssh/sshd_config
rm /etc/ssh/sshd_config.d/60-cloudimg-settings.conf
systemctl restart sshd
```

## Add Guest Agent
```
apt-get update && apt-get -y upgrade
apt-get install -y qemu-guest-agent
```
**reboot**

## Install Docker
```
sh <(curl -sSL https://get.docker.com)
```

## Install Docker Compose
```
LATEST=$(curl -sL https://api.github.com/repos/docker/compose/releases/latest | grep '"tag_name":' | cut -d'"' -f4)
DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
mkdir -p $DOCKER_CONFIG/cli-plugins
curl -sSL https://github.com/docker/compose/releases/download/$LATEST/docker-compose-linux-x86_64 -o ~/.docker/cli-plugins/docker-compose
chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
docker compose version
```

## Add Dockge

- Install Docker Compose first

```
mkdir -p /opt/{dockge,stacks}
wget -q -O /opt/dockge/compose.yaml https://raw.githubusercontent.com/louislam/dockge/master/compose.yaml
cd /opt/dockge
docker compose up -d
```
Dockge GUI, **https:// IP:5001**

## Add Portainer

- Docker run

```
docker volume create portainer_data
docker run -d \
  -p 8000:8000 \
  -p 9443:9443 \
  --name=portainer \
  --restart=always \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v portainer_data:/data \
  portainer/portainer-ce:latest
```

Portainer GUI, **https:// IP:9443**

**Comment by wulungnet** [comment]
Using the Ubuntu 24.04 script.
In **Get SSH Going**, sshd service were named ssh.service now, invoked by ssh.socket [controlled by systemd init].
Additional info : port change can be applied multiple ways but not just by modified sshd_config file only, because init were took control of the start-up. Here you can pick which to choose : [How to change the SSH server port on Ubuntu?](https://serverfault.com/questions/1159599/how-to-change-the-ssh-server-port-on-ubuntu)

**Reply by syl779** [reply]
systemctl restart ssh

**Comment by Braekpo1nt** [comment]
Just putting this here in case someone else has this problem. I had to set Cloud-Init **IP Config (net0)** to DHCP, otherwise I was getting network connectivity issues in the VM. This might be specific to me, but it took me an hour to figure out. In the end, before I clicked "Regenerate image", **IP Config (net0)** was set to "ip=DHCP" in the UI.

**Reply by TimoWolle** [reply]
+1 had the same issue, after switching it to DHCP it worked for me it

**Reply by tristan-k** [reply]
This is still an issue. Thanks for the heads-up.

**Reply by lindstrm** [reply]
+1 but worked with specifying an ip and a gateway as well.

**Reply by its-sven** [reply]
Does anyone have more information about this problem and a way to use Cloud-init correctly to assign the IP address?

**Reply by nwride** [reply]
In my case, i had to configure DHCP on the first bot, and then i was able to change this option to IP address + router gateway

**Comment by sofia-ce** [comment]
Get SSH going option b:
https://gist.github.com/dragonfire1119/8962678cf914e88fe7cccb649c9f5236
(but use systemctl restart ssh, not sshd)

i cant tell you why this above link worked and the commands from this guide didn't D: 

but if there's another soul out there stuck and desperate to ssh, maybe this can help?

**Comment by 2runX** [comment]
![Screenshot 2025-05-20 205044](https://github.com/user-attachments/assets/3e029717-9553-4f53-a775-22595881e141)
![Screenshot 2025-05-20 205058](https://github.com/user-attachments/assets/6b644fd1-edc9-4b86-9e1a-6a5e5e1ee2e1)

**Comment by bcurran3** [comment]
### **If you want to add the GNOME desktop:**

(assuming you've already done `apt update` and are logged into console as root)

Install GNOME desktop only, no apps:
`apt install vanilla-gnome-desktop vanilla-gnome-desktop` 

Install GNOME and core apps only:
`apt install ubuntu-desktop-minimal` 

Install GNOME and additional software such as LibreOffice:
`apt install ubuntu-gnome-desktop ` 

Then you might want GNOME Tweaks for customizing:
`apt install gnome-tweaks`

Once installed, you'll need to reboot the VM.  After which you can access using noVNC.

**Reply by runtimevic** [reply]
Hello,  If I use gnome, the cloud-init password doesn't work....
It was because I had user: root in cloud-init

**Reply by tuxpowered** [reply]
Any suggestions on xfce? (notably debian13  i needed to change to virtio-gpu to get the console to show , but starting xfce (startx) results in on keyboard/mouse but i do get a desktop.

**Comment by sebastienhouzet** [comment]
After installation, what is the default password to access with SSH ?

**Reply by bcurran3** [reply]
> After installation, what is the default password to access with SSH ?

There is no default, it's what you set. Assuming you used the cloud init, scroll up to the first post and look at steps 1 and 2 for a memory jogger.

**Comment by owaismai** [comment]
Corrected code below:

```
sed -i -e 's/#PermitRootLogin prohibit-password/PermitRootLogin yes/g' -e 's/^PasswordAuthentication.*/PasswordAuthentication yes/' /etc/ssh/sshd_config
rm /etc/ssh/sshd_config.d/60-cloudimg-settings.conf
systemctl restart ssh.service
```

**Reply by 2runX** [reply]
thats right

**Comment by Sarthak-Sidhant** [comment]
`/etc/ssh/sshd_config.d/60-cloudimg-settings.conf`
has disabled password authentication

**Comment by ccoles146** [comment]
I don't know why but mine just started automatically. I had to quickly move from pve shell to the new VM and then stop it. Then change the cloud-init settings and then start it.  This worked on the second try, the first one I ended up with a VM with no logon so I just deleted it ;)

**Reply by its-sven** [reply]
You can stop the VM and edit the cloud-init and click `Regenerate Image` so you don't have to delete the VM

**Comment by syl779** [comment]
To add to "Expand VM Disk using parted (/dev/sda1)"

If I completely run out of room, after following the instructions it doesn't always expand - I sometimes find I have to add:

resize2fs /dev/sda1

**Reply by ccoles146** [reply]
You can use the Proxmox GUI to expand the disk and it automatically expands the partition and filesystem, at least in PVE 9. You can then do this in the same step as updating the username and password and you don't need to disable update packages.

**Comment by oldtechie** [comment]
Don't think I'll be using this one again. It completely destroyed my proxmox host completely, not sure how & I am running antivirus everywhere just in case. PRobably just panic.Started all over, I'll build my own VMs in future.

**Comment by axinator** [comment]
`root@ubuntu:~# systemctl restart sshd`
`Failed to restart sshd.service: Unit sshd.service not found.`
`root@ubuntu:~# systemctl restart ssh`

**Comment by 2runX** [comment]
![page_1](https://github.com/user-attachments/assets/3ab5c2bf-28f5-4c96-b78e-3341693e8a5e)
![page_2](https://github.com/user-attachments/assets/6967237c-a54a-4296-a428-224d9affe76d)
![page_3](https://github.com/user-attachments/assets/9eb3be56-be28-4eb5-90a4-ae9657e70898)
